src/main/java/org/czekalski/petapiintegration/config/AnimalRepresentationModelAssembler.java

```diff
@@ -0,0 +1,42 @@
+package org.czekalski.petapiintegration.config;
+
+import org.czekalski.petapiintegration.resourceserver.apiresource.v1.dto.AnimalResourcesListDto;
+import org.czekalski.petapiintegration.resourceserver.controller.AnimalController;
+import org.springframework.hateoas.EntityModel;
+import org.springframework.hateoas.server.mvc.WebMvcLinkBuilder;
+
+import static org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.linkTo;
+import static org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.methodOn;
+
+
+public class AnimalRepresentationModelAssembler {
+
+
+    public EntityModel<AnimalResourcesListDto> toModel(AnimalResourcesListDto animalResourcesListDto, String stateId, String cityId, int size, int page) {
```

Comment:
To jest potworek :) Każdy z tych ciał if elsów powinien być oddzielną klasą. Łatwo wtedy przetestować, czy robią to co mają robić. Dodatkowo jak zastanowisz się nad nazwą takiej klasy, to będziesz rozumiał lepiej, co ona robi. Ten if/else powinien tylko sterować tym, która instancja powinna zostać wykorzystana do zmapowania. Spójrz na pattern factory https://www.tutorialspoint.com/design_pattern/factory_pattern.htm Jak masz więcej pytań, to pisz albo dzwoń :)

Reply:
Problem w: f45d5f1b33b01d46ad17c7f91f352bae0aca3402 Otóż nie mogę przetestować "corner case'ów", ponieważ gdy próbuję to zrobić, czyli dla ostatniej strony zrobić unit test to Spring mi dodaje kolejną stronę do ostatniej strony. Przez co strona na której testuje nigdy nie jest ostatnia. Ta przypadłość jest w klasie AnimalPageCaseFactoryTest

Reply:
Poprawione: 0af844de692538af5d524ca9d17e6776ec2f0316